src/main/java/com/ibm/stocator/fs/swift/SwiftAPIClient.java

```diff
@@ -311,39 +311,40 @@ an request on an object (not a container) that has a trailing slash will lead
       */
       objectNameNoSlash = objectName.substring(0, objectName.length() - 1);
     }
-    StoredObject so = cont.getObject(objectNameNoSlash);
+
+    HttpResponse metadata = SwiftAPIDirect.getObjectMetadata(mJossAccount, container,
+            objectNameNoSlash, swiftConnectionManager);
+
     boolean isDirectory = false;
-    if (so.exists()) {
-      // We need to check if the object size is equal to zero
-      // If so, it might be a directory
-      long contentLength = so.getContentLength();
-      String lastModified = so.getLastModified();
-      if (contentLength == 0) {
-        Collection<DirectoryOrObject> directoryFiles = cont.listDirectory(objectName, '/', "", 10);
-        if (directoryFiles != null && directoryFiles.size() != 0) {
-          // The zero length object is a directory
```

Comment:
@djalova we need to keep this stuff. it actually important part, since Hadoop eco system need notion of directory. The idea here to identify objects of the form "aa/part-001, aa/part-002", etc.. this way aa will be returned as directory. In this case "aa" exists as object of size 0.